ocaml/squeezed/lib/squeeze.mli

```diff
@@ -0,0 +1,247 @@
+val start : float
+
+module D : Debug.DEBUG
+
+val manage_domain_zero : bool ref
+
+val gib : int64
+
+val domain_zero_dynamic_min : int64 ref
+
+val domain_zero_dynamic_max : int64 option ref
+
+val boot_time_host_free_memory_constant_count_min : int ref
+
+val boot_time_host_free_memory_check_interval : float ref
+
+type domain = {
+    domid: int
+  ; can_balloon: bool
+  ; dynamic_min_kib: int64
+  ; target_kib: int64
+  ; dynamic_max_kib: int64
+  ; memory_actual_kib: int64
+  ; memory_max_kib: int64
+  ; inaccuracy_kib: int64
+}
+
+val domain_make :
+  int -> bool -> int64 -> int64 -> int64 -> int64 -> int64 -> int64 -> domain
+
+val domain_to_string_pairs : domain -> (string * string) list
+
+module DomainSet : sig
+  type elt = domain
+
+  type t
+
+  val empty : t
+
+  val is_empty : t -> bool
+
+  val mem : elt -> t -> bool
+
+  val add : elt -> t -> t
+
+  val singleton : elt -> t
+
+  val remove : elt -> t -> t
+
+  val union : t -> t -> t
+
+  val inter : t -> t -> t
+
+  val disjoint : t -> t -> bool
+
+  val diff : t -> t -> t
+
+  val compare : t -> t -> int
+
+  val equal : t -> t -> bool
+
+  val subset : t -> t -> bool
+
+  val iter : (elt -> unit) -> t -> unit
+
+  val map : (elt -> elt) -> t -> t
+
+  val fold : (elt -> 'a -> 'a) -> t -> 'a -> 'a
+
+  val for_all : (elt -> bool) -> t -> bool
+
+  val exists : (elt -> bool) -> t -> bool
+
+  val filter : (elt -> bool) -> t -> t
+
+  val filter_map : (elt -> elt option) -> t -> t
+
+  val partition : (elt -> bool) -> t -> t * t
+
+  val cardinal : t -> int
+
+  val elements : t -> elt list
+
+  val min_elt : t -> elt
+
+  val min_elt_opt : t -> elt option
+
+  val max_elt : t -> elt
+
+  val max_elt_opt : t -> elt option
+
+  val choose : t -> elt
+
+  val choose_opt : t -> elt option
+
+  val split : elt -> t -> t * bool * t
+
+  val find : elt -> t -> elt
+
+  val find_opt : elt -> t -> elt option
+
+  val find_first : (elt -> bool) -> t -> elt
+
+  val find_first_opt : (elt -> bool) -> t -> elt option
+
+  val find_last : (elt -> bool) -> t -> elt
+
+  val find_last_opt : (elt -> bool) -> t -> elt option
+
+  val of_list : elt list -> t
+
+  val to_seq_from : elt -> t -> elt Seq.t
+
+  val to_seq : t -> elt Seq.t
+
+  val to_rev_seq : t -> elt Seq.t
+
+  val add_seq : elt Seq.t -> t -> t
+
+  val of_seq : elt Seq.t -> t
+end
```

Comment:
Repeating the whole signature will cause problems when it's expended in the standard library, better to define it this way:
```suggestion
module DomainSet : Set.S with type elt = domain
```